draft-ietf-masque-connect-udp.md

```diff
@@ -403,18 +428,89 @@ Reference:
 : This document.
 
 
-## Datagram Format Type {#iana-format-type}
+--- back
 
-This document will request IANA to register UDP_PAYLOAD in the "HTTP Datagram
-Format Types" registry established by {{HTTP-DGRAM}}.
+# Example Extensions
 
-|    Type     |   Value   | Specification |
-|:------------|:----------|:--------------|
-| UDP_PAYLOAD | 0xff6f00  | This Document |
-{: #iana-format-type-table title="Registered Datagram Format Type"}
+Extensions can define new semantics for the payload of HTTP Datagrams. The
+extension can then have an endpoint pick an available context ID using the
+allocation service ({{context-id}}) and register that context ID with their
+peer. Registration is the action by which an endpoint informs its peer of the
+semantics and format of a given context ID.
 
 
---- back
+## Registering Contexts with Headers
```

Comment:
I would appreciate some clarification on whether a single extension is expected to offer _both_ registration mechanisms.

If most extensions are expected to support both, I think we should consider adding some standards development guidance to minimize duplicated effort (e.g. registration capsule contents should be exactly the header value).

Reply:
I think some guidance would be useful but it would make most sense to put it alongside the Context ID definition in https://ietf-wg-masque.github.io/draft-ietf-masque-connect-udp/2021_design_team_consensus/draft-ietf-masque-connect-udp.html#section-4

I don't think that needs to block this PR but can instead be done as a follow up WG item.

Reply:
During the last design team meeting, we reached consensus that extension composability should be addressed by the working group as a whole, instead of as part of the design team efforts. When we do look into this as a working group, I'm fairly confident that the appendix about extensions will get rewritten. That'll be the best time to discuss what guidance we want to provide in terms of whether a given extension should offer both registration mechanisms. All this to say I agree with Lucas, let's handle this as a follow up WG item - @bemasc feel free to file an issue to make sure this is tracked.